syntax = "proto2";

message Reader {
  oneof reader_oneof {
    VCRReader vcr_reader = 1;
    VCRTextFRCNNReader vcr_text_frcnn_reader = 2;
  }
}

message VCRReader {
  // Pattern of the input files.
  repeated string input_pattern = 1;

  // Shuffle buffer size.
  optional int32 shuffle_buffer_size = 2 [default = 1000];

  // Cycle length of interleave call.
  optional int32 interleave_cycle_length = 4 [default = 2];

  // Number of parallel calls.
  optional int32 num_parallel_calls = 5 [default = 5];

  // Batch size.
  optional int32 batch_size = 6 [default = 32];

  // Prefetch buffer size.
  optional int32 prefetch_buffer_size = 7 [default = 200];

  // Desired size of the mininum of (height, width).
  optional int32 desired_size = 8 [default = 500];
}

message VCRTextFRCNNReader {
  // Pattern of the input files.
  repeated string input_pattern = 1;

  // Shuffle buffer size.
  optional int32 shuffle_buffer_size = 2 [default = 1000];

  // Cycle length of interleave call.
  optional int32 interleave_cycle_length = 4 [default = 2];

  // Number of parallel calls.
  optional int32 num_parallel_calls = 5 [default = 5];

  // Batch size.
  optional int32 batch_size = 6 [default = 32];

  // Prefetch buffer size.
  optional int32 prefetch_buffer_size = 7 [default = 200];

  // If true, cache dataset in the memory.
  optional bool cache_dataset = 9 [default = false];

  // Dimensions of the Fast-RCNN feature.
  optional int32 frcnn_feature_dims = 10 [default = 1024];
}
